commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/service/PagetabSerializationService.kt

```diff
@@ -23,11 +33,15 @@ internal class PagetabSerializationService(
     override fun deserializeSubmission(content: String, format: SubFormat, source: FilesSource): Submission =
         fileListSerializer.deserializeFileList(serializer.deserializeSubmission(content, format), format, source)
 
-    override fun getSubmissionFormat(file: File) = when (file.extension) {
-        "tsv" -> SubFormat.TSV
-        "xlsx" -> SubFormat.TSV
-        "xml" -> SubFormat.XML
-        "json" -> SubFormat.JSON
-        else -> throw InvalidExtensionException(file)
+    override fun deserializeSubmission(file: File): Submission {
+        return when (fromExtension(file.extension)) {
+            XmlFormat -> deserializeSubmission(file.readText(), XML)
+            is JsonFormat -> deserializeSubmission(file.readText(), JSON)
```

Comment:
Why do we need the `is` here?

Reply:
because there are two json format (normal and parent) so it helps to validate if any of those